engine/CMakeLists.txt

```diff
@@ -84,7 +84,13 @@ set(CUBOS_ENGINE_SOURCE
 	"src/input/bindings.cpp"
 	"src/input/action.cpp"
 	"src/input/axis.cpp"
-	
+
+	"src/ui/canvas/plugin.cpp"
+	"src/ui/canvas/ui_canvas.cpp"
+	"src/ui/canvas/ui_element.cpp"
```

Comment:
You can remove the `ui` prefix from the files, as they're already in a `ui` folder anyway.